src/wasm.rs

```diff
@@ -610,6 +580,75 @@ where
         }
     }
 
+    /// Processes WasmMsg::Instantiate and WasmMsg::Instantiate2 messages.
+    fn process_wasm_msg_instantiate(
+        &self,
+        api: &dyn Api,
+        storage: &mut dyn Storage,
+        router: &dyn CosmosRouter<ExecC = ExecC, QueryC = QueryC>,
+        block: &BlockInfo,
+        sender: Addr,
+        admin: Option<String>,
+        code_id: u64,
+        msg: Binary,
+        funds: Vec<Coin>,
+        label: String,
+        _salt: Option<Binary>,
+    ) -> AnyResult<AppResponse> {
+        if label.is_empty() {
+            bail!("Label is required on all contracts");
+        }
+
+        let contract_addr = self.register_contract(
+            storage,
+            code_id,
+            sender.clone(),
+            admin.map(Addr::unchecked),
+            label,
+            block.height,
+        )?;
```

Comment:
I might be wrong, but I think this behaves different from instantiate2:
For the same sender, checksum / code id and salt, it would generate the same address and fail [here](https://github.com/CosmWasm/wasmd/blob/03f3c72a6ce447fafc2da023a1322899327433f8/x/wasm/keeper/keeper.go#L266), which is not the case with this implementation.
IMHO it's also fine if we ignore that for now and address it in the next step when we have the actual addresses.

Reply:
Good point! I guess I have to make it behave like this way you pointed out. I created an issue #66. Thanks for that tip!